src/main/java/org/fogbowcloud/arrebol/api/http/controllers/JobController.java

```diff
@@ -29,19 +30,29 @@ public JobController(JobService jobService){
     public ResponseEntity<String> addJob(@RequestBody JobSpec jobSpec){
         LOGGER.info("Adding new Job: " + jobSpec + ".");
 
-        String jobId = jobService.addJob(jobSpec);
-        JobResponse jobResponse = new JobResponse(jobId);
+        try{
+            String jobId = jobService.addJob(jobSpec);
+            JobResponse jobResponse = new JobResponse(jobId);
 
-        LOGGER.info("Added " + jobSpec.getLabel() + " with id " + jobId + ".");
-        return new ResponseEntity(jobResponse, HttpStatus.CREATED);
+            LOGGER.info("Added " + jobSpec.getLabel() + " with id " + jobId + ".");
+            return new ResponseEntity(jobResponse, HttpStatus.CREATED);
+        } catch (Exception e){
+            LOGGER.error(String.format(Messages.Exception.GENERIC_EXCEPTION, e.getMessage()), e);
```

Comment:
Also, it would be good to have the jobSpec here is the log

Reply:
(or, at least, the Id/label of th job)

Reply:
Done.